### DISCUSSION TITLE
has anyone here seen nvim-neorg/neorg

### DISCUSSION THREAD

**Discussion by nathanbraun** [nvim-telekasten/telekasten.nvim#59] [General]
Has anyone here seen https://github.com/nvim-neorg/neorg or https://github.com/nvim-neorg/neorg-telescope?  I haven't looked too closely at it them yet, but it looks to be well thought out/have a lot of activity (particularly on the "backend"/syntax/treesitter stuff which maybe complements the link handling/searching portion of telekasten?), but it's unclear exactly how it might relate to this.  The neorg-telescope package in particular looks like it might try to do some of the same things as telekasten, though obviously it's no where close yet.

@renerocksai do you see any future where you might want to someday consider an integration with this format? or do you envision telekasten and these neorg plugins serving different purposes.  Just curious, cheers.

**Comment by renerocksai** [comment]
I have seen neorg, yesterday :smile: - and at first I was impressed, I starred it immediately. I really hope they keep up the pace.

While it looks impressive at first,  they can only link within the same file at the moment, ...

Also, the file format is serving a different purpose. It is bound to become a monster in terms of complexity, in comparison to Markdown files. If you ever looked into org-mode... It's like a million apps for totally different use-cases in one file. I guess you can even do spreadsheet stuff in it, just as an example. [Yep, it can do that](https://orgmode.org/worg/org-tutorials/org-spreadsheet-intro.html). This might lead to a kind-of lock-in. Well, at least there won't be too many tools out there that will support this format. Since their intention is to be incompatible with org-mode, which is understandable, neorg is probably going to be required for neorg-files for a while.

In some regards, neorg reminds me a bit of Roam Research, but with many 'bad' parts solved (no web browser, offline storage). A single 'database' that can contain everything from formatted notes to mermaid diagrams to tomato timers to spreadsheets to kanban boards. The more the better.

Mind you, all of this is super OK - and neorg is fresh and innovative and I really hope they will do the hype in their README justice! Neorg can become super awesome!

Telekasten's main focus, on the other hand, is enabling a digital Zettelkasten (lots of small interlinked notes) in a format that survives decades and is compatible with many, many other tools. Hence, Markdown. I like to keep a journal, too, which blends well with Zettelkasten notes / digital garden stuff, whatever you want to call it. (Well-placed) links make it less important where (in what file) information is stored. 

What I find really cool about neorg is that they have already worked out how to integrate well with treesitter and hence can offer
context-sensitive completions, etc. This is something that I discussed and put into the backlog a 'while' ago but hadn't really dug
into. I guess, treesitter could also be used for re-formatting and re-styling links, for example.

So I guess, neorg is a great source for inspiration for features, polish, etc. 

I can imagine integrating in the following way: enabling links to norg files. Like `[[my-norg]]` which would be stored
in `my-norg.norg` - or at least `[my-norg](my-norg.norg)`. Nobody says you can't have a `.norg` file in your telekasten home. By filetype magic, neorg would take over from there when the link is visited.
